### PR TITLE
Install management-packages with management cluster creation and Add CI to verify CC based cluster creation on AWS

### LINKED ISSUE #2358: Support creating management-cluster using package-based-lcm and with cluster-class
**Describe the feature request**

Request to add support for creating cluster-class based management-cluster when `package-based-lcm` featureflag is enabled.

Currently, we have added support for creation of workload cluster using clusterclass however the managment-cluster gets created in legacy way. We need to add the support for creating management-cluster using cluster-class.

**Describe alternatives you've considered**
none

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- Install management-packages with management cluster creation
- Add CI to verify CC based cluster creation on AWS

### Which issue(s) this PR fixes

Fixes #2358

### Describe testing done for PR
- Using CI to give test signal with verification tests added for the ClusterClass based AWS management-cluster and workload creation

### Release note

```release-note
Install management-packages with management cluster creation and Support creating management-cluster with cluster-class
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by alfredthenarwhal** [comment]
> @anujc25:  /test install-vc7
Commit: e13eab1

Tests failed! Build no: 2179

**Comment by alfredthenarwhal** [comment]
> @rajaskakodkar:  /test install-vc7
Commit: e13eab1

Tests failed! Build no: 2183

**Comment by alfredthenarwhal** [comment]
Tests failed! Build no: 2186

**Comment by alfredthenarwhal** [comment]
Tests passed! Build no: 2192

File: .github/workflows/tkg_integration_tests_clusterclass.yaml [hunk 1]
```diff
@@ -15,17 +15,6 @@ on:
       - '.github/workflows/tkg_integration_tests*.yaml'
       - '.github/workflows/publish_management_package.yaml'
 
-  push:
```
Comment (vuil) [inline .github/workflows/tkg_integration_tests_clusterclass.yaml]: We should publish the packages on a push to main as well.

**Review by vuil — APPROVED** [review]
lgtm, thanks!

**Pull request merged**